dev/format-code.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```

Comment:
What are people's thoughts on merging the `lint`, `clippy` and `cargo-toml-formatting-checks` into a single `lint` check, moving this to script `ci/scripts/rust_lint.sh`, and running it from CI? This will ensure it remains up to date?

@alamb @andygrove ??

Reply:
Personally, I appreciate the fact that multiple steps/commands are visualised as multiple steps/commands in the build pipeline.

The CI build and the dev/lint-code script could be updated to invoke the same scripts (run_clippy.sh, run_xxx, ..) but unless  the commands and/or their arguments change often, I would not bother..

Reply:
Fair enough

Reply:
Definitely an option but you need to be careful about two things though:

* Staging only the files that are relevant - https://github.com/influxdata/influxdb_iox/pull/839/files#diff-87320095d7c4f39eed5d8f3866b512eb910e4eec8e7926faaeef6a53ab786fcbR8
* Avoiding the git hook taking too long

The latter is the really painful one because clippy is very slow, you get no output from git hooks, and they run on most git operations. Formatting could definitely be done in this way though

Reply:
Good point. Why don't we just start with `cargo fmt` for now then. Does that sound reasonable? I feel like those are 80% of the issues that slip through the cracks on commits anyway.

Reply:
I would be quite happy to help those who wanted to install a pre-commit hook but am opposed to anything that required it / made it the default in this repository as various developers have their own ideas of ideal workflows

I suggest adding the three scripts as suggested
```
ci/scripts/rust_fmt.sh
ci/scripts/rust_clippy.sh
ci/scripts/toml_fmt.sh
```
that are invoked from the CI jobs (and thus remain in sync)

And then people can mix/match however they want to run them

Reply:
Created the scripts
Updated rust.yml to call the scripts
Updated (and renamed) dev/rust_lint.sh to call the scripts

Reply:
Added branches to verify that build still fails when one of the checks is not ok:

* https://github.com/timvw/arrow-datafusion/tree/ISSUE-2321-break-fmt -> build caught issue: https://github.com/timvw/arrow-datafusion/actions/runs/2276886712
 
* https://github.com/timvw/arrow-datafusion/tree/ISSUE-2321-break-toml -> build caught issue:
https://github.com/timvw/arrow-datafusion/actions/runs/2276894979

* https://github.com/timvw/arrow-datafusion/tree/ISSUE-2321-break-clippy -> build caught issue:
https://github.com/timvw/arrow-datafusion/actions/runs/2276926769